tools/batchTestGenerator/cmd/root.go

```diff
@@ -0,0 +1,172 @@
+			if c.runConfig.OutputLocation == "" {
+				//validate input/output set to cwd if necessary
+				defaultFileLoc, err := os.Getwd()
+				if err != nil {
+					return fmt.Errorf("failed to get cwd: %w", err)
+				}
+				c.runConfig.OutputLocation = defaultFileLoc
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return batch.LocalGenerator(c.runConfig)
+		},
+	}
+
+	c.githubCommand = cobra.Command{
+		Use:   "github",
+		Short: "Genereate two outputs for GitHub Actions",
+		Long: `This command generates a batch-test-key and batch-test-values
+		output for use in GitHub Actions. See README for example usage.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return batch.GithubGenerator(c.runConfig)
+		},
+	}
+	c.rootCommand.AddCommand(&c.localCommand)
+	c.rootCommand.AddCommand(&c.githubCommand)
+
+	return c
+}
+
+var (
+	comCfg = newCommandConfig()
+)
+
+func Execute() {
+	err := comCfg.rootCommand.Execute()
+	if err != nil {
+		log.Printf("failed execute: %v", err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	// Persistent Flags
+	comCfg.rootCommand.PersistentFlags().StringVar(&comCfg.runConfig.TestCaseFilePath, "testCaseFilePath", "", `path to JSON test case file`)
+	comCfg.rootCommand.PersistentFlags().StringSliceVar(&comCfg.includeFlags, "include", []string{}, "list of commma separated services to include. See README for list of valid values.")
+	comCfg.rootCommand.PersistentFlags().StringVar(&comCfg.eksARM64Flags.ampEndpoint, "eksarm64amp", "", "AMP endpoint for EKS ARM64 tests")
+	comCfg.rootCommand.PersistentFlags().StringVar(&comCfg.eksARM64Flags.clusterName, "eksarm64cluster", "", "Cluster name for EKS ARM64 tests")
+	comCfg.rootCommand.PersistentFlags().StringVar(&comCfg.eksARM64Flags.region, "eksarm64region", "", "Region for EKS ARM64 tests")
+	comCfg.rootCommand.PersistentFlags().StringVar(&comCfg.eksFlags.ampEndpoint, "eksamp", "", "AMP endpoint for EKS tests")
+	comCfg.rootCommand.PersistentFlags().StringVar(&comCfg.eksFlags.clusterName, "ekscluster", "", "Cluster name for EKS tests")
+	comCfg.rootCommand.PersistentFlags().StringVar(&comCfg.eksFlags.region, "eksregion", "", "Region for EKS tests")
+
+	// githubflags only
+	comCfg.githubCommand.Flags().IntVar(&comCfg.runConfig.MaxBatches, "maxBatch", 40, "Maxium number of batches allowed.")
+
+	//local flags only
+	comCfg.localCommand.Flags().StringVar(&comCfg.runConfig.OutputLocation, "output", "", "Output location for test-case-batch file.")
+	comCfg.localCommand.Flags().IntVar(&comCfg.runConfig.MaxBatches, "maxJobs", 5, "Maximum number of jobs allowed in test-case-batch file. Will generate tests up to this amount if possible from"+
+		" provided test cases and included services. ")
+
+}
+
+// transform array slice into map
+func transformInclude(includedFiles []string) map[string]struct{} {
+	output := make(map[string]struct{})
+	for _, val := range includedFiles {
+		output[val] = struct{}{}
```

Comment:
Should this have an allowlist for validation?

Reply:
Yes, that's a good idea. I'll add one.

Reply:
Added